### ISSUE TITLE
webpack-dev-server warning in html-entities

### ISSUE THREAD

**Issue by Samalve** [webpack/webpack-dev-server#3808]
- [x] This is a **bugFix**

### Code

```js
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
  entry: './src/index.js',
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'bundle.[contenthash].js',
    clean: true,
  },
  resolve: {
    extensions: ['.js', '.jsx'],
  },
  module: {
    rules: [
      {
        test: /\.(js|jsx)/,
        exclude: /node_modules/,
        loader: 'babel-loader',
      },
      {
        test: /.html/,
        loader: 'html-loader',
      },
    ],
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: path.join(__dirname, 'public/index.html'),
      filename: 'index.html',
      hash: true,
      inject: 'body',
    }),
  ],
  devServer: {
    static: {
      directory: path.join(__dirname, 'public/index.html'),
    },
    port: 9000,
  },
};
```

### Expected Behavior
Let webpack compile as normal

### Actual Behavior
![image](https://user-images.githubusercontent.com/42875378/132405410-2d3e542d-be14-478c-9035-b7d08a1f33e8.png)

### For Bugs; How can we reproduce the behavior?
The bug occurs when compiling webpack in development mode in ```./node_modules/webpack-dev-server/client/overlay.js```

### For Features; What is the motivation and/or use-case for the feature?
The bug is solved by removing the exact import
![image](https://user-images.githubusercontent.com/42875378/132406086-7bf23274-5e56-4d40-913b-eb88845458e8.png)
by default one
![image](https://user-images.githubusercontent.com/42875378/132406133-35ee0d4b-f1e4-4e37-ad92-9ae1b48df57e.png)

**Comment by alexander-akait** [comment]
Expected and was answered a lot of time,please use:
```
 resolve: {
    extensions: ['.jsx', '...'],
  },
```

**Comment by Samalve** [comment]
the error still occurs
![image](https://user-images.githubusercontent.com/42875378/132406908-9e2ac421-ad7a-4608-b005-bc1892c055e3.png)

**Comment by alexander-akait** [comment]
It means you still change `resolve` options somewhere else, open `html-entities` https://github.com/mdevils/html-entities/blob/master/src/index.ts#L73 and you will find `encode` is named export

**Comment by alexander-akait** [comment]
And from usage https://github.com/mdevils/html-entities#decodetext-options

**Comment by Samalve** [comment]
how strange for me it only worked by replacing the import
I share my [repo](https://github.com/Samalve/todo-app)

**Comment by Samalve** [comment]
And it also throws me this error 
```
Uncaught TypeError: ansi_html_community__WEBPACK_IMPORTED_MODULE_0__.default.setColors is not a function
    at eval (overlay.js:27)
    at Module../node_modules/webpack-dev-server/client/overlay.js (bundle.44e4d49b9a5376169473.js?0677c222938670ff7226:2)
    at __webpack_require__ (bundle.44e4d49b9a5376169473.js?0677c222938670ff7226:2)
    at s (bundle.44e4d49b9a5376169473.js?0677c222938670ff7226:2)
    at eval (index.js:9)
    at Module../node_modules/webpack-dev-server/client/index.js?protocol=ws%3A&hostname=0.0.0.0&port=9000&pathname=%2Fws&logging=info (bundle.44e4d49b9a5376169473.js?0677c222938670ff7226:2)
    at __webpack_require__ (bundle.44e4d49b9a5376169473.js?0677c222938670ff7226:2)
    at bundle.44e4d49b9a5376169473.js?0677c222938670ff7226:2
    at bundle.44e4d49b9a5376169473.js?0677c222938670ff7226:2
```

**Comment by alexander-akait** [comment]
Note - `file-loader` and `url-loader` deprecated  https://webpack.js.org/guides/asset-modules/

**Comment by alexander-akait** [comment]
But problem here `  test: /.html/,`, you should use `  test: /.html$/,`, becuase `/.html/` pass `node_modules/html-entities/`

**Comment by alexander-akait** [comment]
Even more please use `$` for all `test` where used `extension`

**Comment by lukethacoder** [comment]
Ran into a similar issue. I removed my entire `resolve.extensions` config and it seemed to start working without any issues.

I assume this isn't possible for all users. I am using plugins (`lwc-webpack-plugin` and `html-webpack-plugin`) to handle the imports instead of the resolve config + loaders in webpack.

Still odd that it would trip up on the `html-entities` import and the `ansiHTML.setColors(colors)` method (from the `ansi-html-community` package).
